build/tools/utils.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+function check_deps() {
+	local dependencies=("$@")
+	echo "Checking dependencies..."
+
+	for dep in ${dependencies[@]}; do
+	if ! command -v $dep &> /dev/null; then
+		echo "Please install $dep"
+		exit 1
+	fi
+	done
+
+	echo "All dependencies are installed"
+}
+
+function strip_prefix() {
+	local str="$1"
+	local prefix="$2"
+
+	echo ${str#"$prefix"}
+}
+
+function version_without_v() {
+	strip_prefix "$1" v
+}
+
+function version_with_v() {
+	local without_v=$(version_without_v "$1")
+
+	echo "v$without_v"
+}
+
+
+# get_noobaa_version returns the NooBaa version by reading the `cmd/version/main.go`
+#
+# Note: If the second argument is set to 0, it will strip "v" from the version (default)
+# Note: If the second argument is set to 1, it will append "v" to the version
+function get_noobaa_version() {
+  local version=$(go run cmd/version/main.go)
+  if [[ $# -eq 1 ]]; then
+	if [[ "$1" == "1" ]]; then
+		version_with_v "$version"
+	else
+		version_without_v "$version"
+	fi
+  else
+    version_without_v "$version"
+  fi
+}
```

Comment:
not sure that I like 1 and 0:
0 has no meaning here as we don't even check that so the comment is misleading.
and using $1 directly should be avoided
I suggest using a variable and changing it if we get the argument:
maybe something like: 
```suggestion
# get_noobaa_version returns the NooBaa version by reading the `cmd/version/main.go`
#
# Note: If an argument is presented, it will append "v" to the version
function get_noobaa_version() {
  local version=$(go run cmd/version/main.go)
  if [[ $# -eq 1 ]]
  then
	version_with_v "$version"
  else
    version_without_v "$version"
  fi
}
```

Reply:
I think the comments are stale, previously this function explicitly just supported `0` (false) and `1` (true).

Will commit the above suggestion.

Reply:
I didn't notice this yesterday but I am not sure if this is a good function signature. It allows the argument to be completely arbitrary which seems a bit odd (it might be that I am just being pedantic).

Reply:
Yes, that's true, that's the reason for `maybe something like:` 😄 
The thing is that in bash you need to check it, but in this case, ANY flag should do it.
Well, the example above is if you get a flag act this way otherwise act as the default.
If you want to be strict you can get the $1 into a variable and check its value, and prompt an error if that is not what you expected.
might be overkill as this function is internal, but you can do it.